internal/transport/http2_server.go

```diff
@@ -521,14 +521,15 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 	}
 	if httpMethod != http.MethodPost {
 		t.mu.Unlock()
+		errMsg := fmt.Sprintf("http2Server.operateHeaders parsed a :method field: %v which should be POST", httpMethod)
 		if logger.V(logLevel) {
-			logger.Infof("transport: http2Server.operateHeaders parsed a :method field: %v which should be POST", httpMethod)
+			logger.Infof("transport: %v", errMsg)
 		}
-		t.controlBuf.put(&cleanupStream{
-			streamID: streamID,
-			rst:      true,
-			rstCode:  http2.ErrCodeProtocol,
-			onWrite:  func() {},
+		t.controlBuf.put(&earlyAbortStream{
```

Comment:
I believe we'll still want to do a RST_STREAM for this stream with NO_ERROR, unless the client has set the END_STREAM flag on this headers frame.  This is what we do for regularly established streams when we end them (RST_STREAM unless client has half-closed already).  E.g.

https://github.com/grpc/grpc-go/blob/333a441e27395b34f666e7d8f0ba871fae2ed53b/internal/transport/http2_server.go#L1061-L1062

Should `earlyAbortStream` just take care of that for us, maybe (optionally based on a boolean presumably)?